packages/core/src/api/thread.ts

```diff
@@ -0,0 +1,117 @@
+import type { RawFile, REST } from '@discordjs/rest';
+import {
+	Routes,
+	type APIThreadChannel,
+	type APIThreadMember,
+	type RESTGetAPIChannelThreadMembersResult,
+	type RESTPostAPIChannelThreadsJSONBody,
+	type RESTPostAPIChannelThreadsResult,
+	type RESTPostAPIGuildForumThreadsJSONBody,
+	type Snowflake,
+} from 'discord-api-types/v10';
+
+export interface StartThreadOptions extends RESTPostAPIChannelThreadsJSONBody {
+	message_id?: string;
+}
+
+export interface StartForumThreadOptions extends RESTPostAPIGuildForumThreadsJSONBody {
+	message: RESTPostAPIGuildForumThreadsJSONBody['message'] & { files?: RawFile[] };
+}
+
+export class ThreadsAPI {
+	public constructor(private readonly rest: REST) {}
+
+	/**
+	 * Fetches a thread
+	 *
+	 * @param channelId - The id of the channel to fetch the thread from
+	 * @param threadId - The id of the thread
+	 */
+	public async get(channelId: Snowflake, threadId: Snowflake) {
+		return this.rest.get(Routes.threads(channelId, threadId)) as Promise<APIThreadChannel>;
+	}
+
+	/**
+	 * Creates a new thread
+	 *
+	 * @param channelId - The id of the channel to start the thread in
+	 * @param options - The options to use when starting the thread
+	 */
+	public async start(channelId: Snowflake, { message_id, ...body }: StartThreadOptions) {
```

Comment:
The documentation and the method name don't quite match, shouldn't this be named `create`?